react/features/virtual-background/components/VirtualBackgroundDialog.js

```diff
@@ -67,42 +67,77 @@ function VirtualBackground({ dispatch, t }: Props) {
      * Updates stored images on local storage.
      */
     useEffect(() => {
-        jitsiLocalStorage.setItem('virtualBackgrounds', JSON.stringify(storedImages));
+        try {
+            jitsiLocalStorage.setItem('virtualBackgrounds', JSON.stringify(storedImages));
+        } catch (err) {
+            // Preventing localStorage QUOTA_EXCEEDED_ERR
+            err && deleteStoredImage(storedImages[0]);
+        }
         if (storedImages.length === backgroundsLimit) {
             deleteStoredImage(storedImages[0]);
         }
     }, [ storedImages ]);
 
     const [ selected, setSelected ] = useState('');
-    const enableBlur = async () => {
+    const enableBlur = async (blurValue, selection) => {
         isloading(true);
-        setSelected('blur');
-        await dispatch(setVirtualBackground('', false));
-        await dispatch(toggleBackgroundEffect(true));
+        setSelected(selection);
+        await dispatch(
+            toggleBackgroundEffect({
+                enabled: true,
+                blurValue,
+                virtualBackground: {
+                    url: ''
+                }
+            })
+        );
         isloading(false);
     };
 
     const removeBackground = async () => {
         isloading(true);
         setSelected('none');
-        await dispatch(setVirtualBackground('', false));
-        await dispatch(toggleBackgroundEffect(false));
+        await dispatch(
+            toggleBackgroundEffect({
+                enabled: false,
+                blurValue: 0,
+                virtualBackground: {
+                    url: ''
+                }
+            })
+        );
         isloading(false);
     };
 
     const setUploadedImageBackground = async image => {
         isloading(true);
         setSelected(image.id);
-        await dispatch(setVirtualBackground(image.src, true));
-        await dispatch(toggleBackgroundEffect(true));
+        await dispatch(
+            toggleBackgroundEffect({
+                enabled: true,
+                blurValue: 0,
+                virtualBackground: {
```

Comment:
This looks very odd. Why do we have a nested object just for a single option? Also we are passing the blue value when using the image and vice versa. You can add a "type" option which can be "blur" or "image" and then only set the options that apply (url or blur value respectively).

Reply:
Simplified the structure